GG/GG/Font.h

```diff
@@ -233,6 +241,40 @@ class GG_API Font
         mutable X cached_width;
     };
 
+    /** \brief Used to encapsulate a text and tokens together to be rendered
+        using GG::Font. */
+    class GG_API TextElements
+    {
+    public:
+        TextElements(const Font& font);
+        ~TextElements();
+
+        /** Return the constructed text.*/
+        const std::string& Text();
+        /** Return the constructed TextElements.*/
+        const std::vector<boost::shared_ptr<TextElement> >& Elements();
```

Comment:
I think that the TextElement is a tree-like structure. When having a formatted string like:

`The <i>quick <rgba 102 51 0 255>brown</rgba></i> fox jumps over the <b>lazy<b> dog.`

Would create something like:

![foo](https://cloud.githubusercontent.com/assets/84891/18948117/144b42fa-8636-11e6-9c02-b02d8f4ce78d.png)